src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/RSACryptoServiceProvider.cs

```diff
@@ -132,17 +206,7 @@ public bool PersistKeyInCsp
         {
             get
             {
-                if (_safeProvHandle == null)
-                {
-                    lock (this)
-                    {
-                        if (_safeProvHandle == null)
-                        {
-                            _safeProvHandle = CapiHelper.CreateProvHandle(_parameters, _randomKeyContainer);
-                        }
-                    }
-                }
-                return CapiHelper.GetPersistKeyInCsp(_safeProvHandle);
+                return CapiHelper.GetPersistKeyInCsp(SafeProvHandle);
```

Comment:
My take on it is that the locking was to ensure that none of the SafeHandles ever got finalized, so that was the only lock semantic I was preserving.